daemon/daemon.go

```diff
@@ -1595,3 +1595,19 @@ func (i *imageBackend) GetRepository(ctx context.Context, ref reference.Named, a
 		},
 	})
 }
+
+// GetRepositories returns a list of repositories configured for the given
+// reference. Multiple repositories can be returned if the reference is for
+// the default (Docker Hub) registry and a mirror is configured, but it omits
+// registries that were not reachable (pinging the /v2/ endpoint failed).
+//
+// It returns an error if it was unable to reach any of the registries for
+// the given reference, or if the provided reference is invalid.
+func (i *imageBackend) GetRepositories(ctx context.Context, ref reference.Named, authConfig *registrytypes.AuthConfig) ([]dist.Repository, error) {
```

Comment:
Just so I understand, this one isn't used anywhere right? I'm not sure what the executor.ImageBackend interface is used for

Reply:
This one is used in the router (as `backend.GetRepositories()`).

The `executor.ImageBackend` interface is used for swarm services. I have a follow-up to update that one to use this as well (instead of the `GetRepository()` (singular), but it's only used for fallback cases on old < 1.30 API versions. That made it less urgent (if we want to fix it at all!) so I decided to keep that commit separate.

I'll push a draft PR with that follow-up.

Reply:
Pushed the follow-up as draft;

- https://github.com/moby/moby/pull/47024